### PR TITLE
UNO: Make autostart command

### PR DESCRIPTION
**Pull request by HarmanPC**

### CODE REVIEW

**Comment by whalemer** [comment]
This needs to be cleared when the game is ended

File: chat-plugins/uno.js [hunk 1]
```diff
@@ -596,6 +597,23 @@ exports.commands = {
...
+		autostart: function (target, room, user) {
+			if (!this.can('minigame', null, room)) return;
+			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			let amount = parseInt(target);
+			if (!amount || amount < 30 || amount > 600) return this.errorReply("The amount must be a number between 30 and 600.");
+			room.game.autostartTimer = amount;
+			if (room.game.autostartTimer) {
+				clearTimeout(room.game.autostartTimer);
+			}
+			room.game.autostartTimer = setTimeout(() => {
```
Comment (whalemer) [inline chat-plugins/uno.js]: imo it's cleaner just to clear it in destroy() and onstart()

Comment (HarmanPC) [reply chat-plugins/uno.js]: oh

**Comment by whalemer** [comment]
you're still able to set an autostart timer after the game's started

``if (room.game.state !== signups) return this.errorReply(whatever)``

File: chat-plugins/uno.js [hunk 2]
```diff
@@ -596,6 +599,22 @@ exports.commands = {
 			this.modlog('UNO TIMER', null, `${amount} seconds`);
 		},
 
+		autostart: function (target, room, user) {
+			if (!this.can('minigame', null, room)) return;
+			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			let amount = parseInt(target);
```
Comment (whalemer) [inline chat-plugins/uno.js]: while im being super nitpicky, ``const``

**Comment by HarmanPC** [comment]
beep boop

**Comment by Zarel** [comment]
idk, someone want to review

File: chat-plugins/uno.js [hunk 3]
```diff
@@ -596,6 +599,23 @@ exports.commands = {
 			this.modlog('UNO TIMER', null, `${amount} seconds`);
 		},
 
+		autostart: function (target, room, user) {
+			if (!this.can('minigame', null, room)) return;
+			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			const amount = parseInt(target);
+			if (!amount || amount < 30 || amount > 600) return this.errorReply("The amount must be a number between 30 and 600.");
+			if (room.game.state !== 'signups') return this.errorReply("The game of UNO has already started.");
+			room.game.autostartTimer = amount;
```
Comment (whalemer) [inline chat-plugins/uno.js]: why

File: chat-plugins/uno.js [hunk 4]
```diff
@@ -596,6 +599,23 @@ exports.commands = {
...
+			if (room.game.autostartTimer) {
```
Comment (whalemer) [inline chat-plugins/uno.js]: pretty nitpicky but i dont see why this cant be one line

**Comment by whalemer** [comment]
might want a way to remove an existing timer

File: chat-plugins/uno.js (hunk 3 above)
Comment (HarmanPC) [reply chat-plugins/uno.js]: ?

**Comment by bumbadadabum** [comment]
Does this really need to be modnoted?

**Comment by HarmanPC** [comment]
maybe

**Comment by whalemer** [comment]
tours do, but im not sure if uno is important enough for people to care

**Comment by HarmanPC** [comment]
Should I remove adding log then?

**Comment by HarmanPC** [comment]
hm?

**Comment by HoeenCoder** [comment]
yeah id stop modlogging

**Comment by AustinXII** [comment]
@Zarel Sparkychild created it but died so I guess it’s up for grabs afaik

**Comment by HarmanPC** [comment]
o.o

**Comment by HarmanPC** [comment]
Merge when

**Comment by HarmanPC** [comment]
@Zarel

**Comment by Zarel** [comment]
Find another coder to approve this; I don't have the time to.

File: chat-plugins/uno.js [hunk 5]
```diff
@@ -596,6 +599,23 @@ exports.commands = {
 			this.modlog('UNO TIMER', null, `${amount} seconds`);
 		},
 
+		autostart: function (target, room, user) {
+			if (!this.can('minigame', null, room)) return;
+			if (!room.game || room.game.gameid !== 'uno') return this.errorReply("There is no UNO game going on in this room right now.");
+			if (toId(target) === 'off') {
+				if (!room.game.autostartTimer) return this.errorReply("There is no autostart timer running on.");
+				this.addModAction(`${user.name} has turned off UNO autostart timer.`);
```
Comment (bumbadadabum) [inline chat-plugins/uno.js]: grammar

File: chat-plugins/uno.js [hunk 6]
```diff
@@ -596,6 +599,23 @@ exports.commands = {
...
+				clearTimeout(room.game.autostartTimer);
+				return;
+			}
+			const amount = parseInt(target);
+			if (!amount || amount < 30 || amount > 600) return this.errorReply("The amount must be a number between 30 and 600.");
```
Comment (bumbadadabum) [inline chat-plugins/uno.js]: Specify you're talking about seconds.

File: chat-plugins/uno.js [hunk 7]
```diff
@@ -596,6 +599,23 @@ exports.commands = {
...
+			if (room.game.state !== 'signups') return this.errorReply("The game of UNO has already started.");
+			if (room.game.autostartTimer) clearTimeout(room.game.autostartTimer);
+			room.game.autostartTimer = setTimeout(room.game.onStart, amount * 1000);
+			this.addModAction(`${user.name} has set the autostart timer to ${amount} seconds.`);
```
Comment (bumbadadabum) [inline chat-plugins/uno.js]: Specify this is for the UNO game.

**Comment by Zarel** [comment]
```
The main process crashed again with this stack trace:
TypeError: Cannot read property 'gameNumber' of undefined
    at Timeout.onStart [as _onTimeout] (/home/ps/showdown/chat-plugins/uno.js:108:48)
    at ontimeout (timers.js:469:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:264:5)
```

I am in zero way surprised by this.

**Pull request merged**